packages/medusa/src/api/routes/store/products/list-products.ts

```diff
@@ -303,6 +313,15 @@ export class StoreGetProductsParams extends StoreGetProductsPaginationParams {
   @FeatureFlagDecorators(SalesChannelFeatureFlag.key, [IsOptional(), IsArray()])
   sales_channel_id?: string[]
 
+  @IsArray()
+  @IsOptional()
+  category_id?: string[]
+
+  @IsBoolean()
+  @IsOptional()
+  @Transform(({ value }) => optionalBooleanMapper.get(value.toLowerCase()))
+  category_children?: boolean
```

Comment:
nit,suggestion(non-blocking): Since this is a boolean flag, we could make it more intuitive by renaming it to `include_category_children` or similar